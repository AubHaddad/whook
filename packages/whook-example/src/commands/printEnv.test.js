import initPrintEnvCommand from './printEnv';

describe('printEnvCommand', () => {
  const log = jest.fn();

  beforeEach(() => {
    log.mockReset();
  });

  it('should work', async () => {
    const printEnvCommand = await initPrintEnvCommand({
      log,
      ENV: { NODE_ENV: 'test' },
      args: {
        _: ['env'],
        keysOnly: true,
      },
    });
    const result = await printEnvCommand();

    expect({
      result,
      logCalls: log.mock.calls.filter(args => 'stack' !== args[0]),
    }).toMatchSnapshot();
  });
});
