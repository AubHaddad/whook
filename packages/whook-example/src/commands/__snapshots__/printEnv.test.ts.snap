// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`printEnvCommand should work 1`] = `
{
  "logCalls": [
    [
      "info",
      "["NODE_ENV"]",
    ],
  ],
  "result": undefined,
}
`;
